containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/TokenAuthenticator.java

```diff
@@ -490,17 +487,18 @@ protected void stringDone(String body) {
                     boolean allowed = allowedNode != null && allowedNode.asBoolean();
                     onRequestResult(serverExchange, connection, allowed);
                 } catch (IOException e) {
-                    onRequestFailure(serverExchange, connection, e);
+                    onRequestFailure(serverExchange, connection, e, true);
                 }
             } else {
                 IOException e = new IOException(StatusCodes.getReason(clientExchange.getResponse().getResponseCode()));
-                onRequestFailure(serverExchange, connection, e);
+                // Retries will not help to solve the issue
+                onRequestFailure(serverExchange, connection, e, false);
```

Comment:
I don't think we should worry too much about what the current kubernetes implementation is doing, we might not even be hitting the exact endpoint but some proxy or a load balancer.

I think this should just be done with best effort and based on the definitions of what those status codes typically means (eg 503 typically is a temporary problem, but not always). Even if we just retry with a 503 and not retry with the others I think would be a good solution.

If we have a need to tweak this later if deemed necessary.